src/lib.rs

```diff
@@ -27,142 +29,110 @@ pub mod payloads {
+                    }
+                    None => {
+                        tracing::warn!(target: BLOCKSAPI, "End of stream reached");
+                        anyhow::bail!("End of stream reached");
+                    }
+                }
+            }
         }
-        _ => Err(anyhow::anyhow!(
-            "Unknown version of borealis-message: {}",
-            version
-        )),
     }
 }
 
-/// Decode the Borealis payload containing LZ4 compressed JSON data
-/// The payload is expected to be in the format:
-/// [1-byte version][BorealisEnvelope][LZ4 compressed JSON bytes]
-/// Returns the decompressed JSON bytes
-async fn decode_near_block_json(data: &[u8]) -> anyhow::Result<Vec<u8>> {
-    // First decode the borealis payload to get the LZ4 compressed data
-    let lz4_data: Vec<u8> = decode_borealis_payload(data).await?;
-    let mut buf = vec![];
-    lz4::Decoder::new(std::io::Cursor::new(lz4_data))?.read_to_end(&mut buf)?;
-    Ok(buf)
-}
-
-/// Patch the JSON to add missing fields for compatibility with near-indexer-primitives
-/// Specificaly, add "chunks": [] if missing in the block object
-/// This is needed because near-indexer-primitives expects the "chunks" field to be present
-/// even if it's empty
-async fn patch_streamer_message_json(json_data: Vec<u8>) -> anyhow::Result<serde_json::Value> {
-    let mut json_msg = serde_json::from_slice::<serde_json::Value>(&json_data)?;
-    if let Some(block) = json_msg.get_mut("block") {
-        if let Some(block_obj) = block.as_object_mut() {
-            // Add chunks field as empty array if it doesn't exist
-            block_obj
-                .entry("chunks")
-                .or_insert_with(|| serde_json::Value::Array(vec![]));
+/// Start streaming blocks from the Blocks API server based on the provided configuration.
+/// Blocks are sent to the provided `streamer_message_sink` channel.
+/// This function manages the streaming, including handling catch-up logic and
+/// ensuring blocks are processed in order.
+async fn start(
+    streamer_message_sink: mpsc::Sender<near_indexer_primitives::StreamerMessage>,
+    config: BlocksApiConfig,
+) -> anyhow::Result<()> {
+    // Spawn a task to continuously update the final block height
+    let final_block_height_atomic = std::sync::Arc::new(std::sync::atomic::AtomicU64::new(0));
+    tokio::spawn(task_update_final_block_regularly(
+        final_block_height_atomic.clone(),
+        config.clone(),
+    ));
+
+    let mut final_block_height =
+        final_block_height_atomic.load(std::sync::atomic::Ordering::SeqCst);
+
+    // Wait for the final block height to be updated if it's still 0
+    if final_block_height == 0 {
+        tracing::info!(target: BLOCKSAPI, "Waiting for final block height to be available...");
+        let mut retries = 0;
+        while final_block_height == 0 && retries < 5 {
+            tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
```

Comment:
Please use const at the top to avoid magic numbers. Don't expose it but just make it more maintainable